suppression-backup-service/exporter/exporter.go

```diff
@@ -0,0 +1,223 @@
+	err := os.MkdirAll(baseDir, os.ModePerm)
+	if err != nil {
+		return nil, fmt.Errorf("could not create tmp dir: %w", err)
+	}
+	repository, err := suppression.NewBadgerRepository(
+		baseDir,
+		pkgLogger)
+	if err != nil {
+		return nil, fmt.Errorf("could not create badger repository: %w", err)
+	}
+	return repository, nil
+}
+
+type Exporter struct {
+	id             identity.Identifier
+	file           model.File
+	log            logger.Logger
+	pollIntervalFn func() time.Duration
+}
+
+// WithPollIntervalFn sets the interval at which the syncer will poll the backend
+func WithPollIntervalFn(pollIntervalFn func() time.Duration) ExporterOpt {
+	return func(c *Exporter) {
+		c.pollIntervalFn = pollIntervalFn
+	}
+}
+
+// NewExporter creates a new syncer
+func NewExporter(id identity.Identifier, file model.File, log logger.Logger, opts ...ExporterOpt) (*Exporter, error) {
+	e := &Exporter{
+		id:   id,
+		file: file,
+		log:  log,
+		pollIntervalFn: func() time.Duration {
+			return time.Hour
+		},
+	}
+	for _, opt := range opts {
+		opt(e)
+	}
+	return e, nil
+}
+
+func (e *Exporter) FullExporterLoop(ctx context.Context) error {
+	tmpDir, err := misc.CreateTMPDIR()
+	if err != nil {
+		return err
+	}
+	repo, err := newBadgerDBInstance(path.Join(tmpDir, "fullsuppression"), e.log)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = repo.Stop()
+	}()
+
+	syncer, err := suppression.NewSyncer(
+		config.GetString("SUPPRESS_USER_BACKEND_URL", "https://api.rudderstack.com"),
+		e.id,
+		repo,
+		suppression.WithLogger(e.log),
+		suppression.WithHttpClient(&http.Client{Timeout: config.GetDuration("HttpClient.suppressUser.timeout", 30, time.Second)}),
+		suppression.WithPageSize(config.GetInt("BackendConfig.Regulations.pageSize", 5000)),
+	)
+	if err != nil {
+		return err
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+			syncStart := time.Now()
+			if err := syncer.Sync(ctx); err != nil {
+				return err
+			}
+			stats.Default.NewStat("suppression_backup_service_full_sync_time", stats.TimerType).Since(syncStart)
+			exportStart := time.Now()
+			if err = Export(repo, e.file); err != nil {
+				return err
+			}
+			stats.Default.NewStat("suppression_backup_service_full_export_time", stats.TimerType).Since(exportStart)
+			<-time.After(e.pollIntervalFn())
+		}
+	}
+}
+
+func (e *Exporter) LatestExporterLoop(ctx context.Context) error {
+	var interval time.Duration
+	config.RegisterDurationConfigVariable(600, &interval, true, time.Second, "SuppressionExporter.latestExportInterval")
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+			latestExport := func() error {
+				tmpDir, err := misc.CreateTMPDIR()
+				if err != nil {
+					return err
+				}
+				baseDir := path.Join(tmpDir, "latestsuppression")
+				repo, err := newBadgerDBInstance(baseDir, e.log)
+				if err != nil {
+					return err
+				}
+				defer func() {
+					_ = repo.Stop()
+					_ = os.RemoveAll(baseDir)
```

Comment:
This might be redundant. We definitely need to make sure that folder doesn't exist before starting the badgerDB instead.

Reply:
removed `os.RemoveAll` from here to before calling `newBadgerDBInstance`